lib/canvas/live_events.rb

```diff
@@ -101,6 +101,23 @@ def self.get_submission_data(submission)
     }
   end
 
+  def self.get_attachment_data(attachment)
+    {
+      attachment_id: attachment.global_id,
+      user_id: attachment.global_user_id,
+      display_name: LiveEvents.truncate(attachment.display_name),
+      filename: LiveEvents.truncate(attachment.filename),
+      context_type: attachment.context_type,
+      context_id: attachment.context_id,
```

Comment:
The `context_type` and `context_id` included in the `metadata` body is only there if the event was triggered by a web request, and will be the context for the request itself, which is likely not what you want. Including these here appear to be what you want.